src/transformState.js

```diff
@@ -5,7 +5,24 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        };
+        break;
+
+      case 'clear':
+        for (const property of Object.keys(state)) {
+          delete state[property];
+        }
+    }
```

Comment:
4. [CHECKLIST] [CODE STYLE]: switch/case should always have default case for error handling.